CONTRIBUTING.md

```diff
@@ -294,3 +289,8 @@ This quality level is meant for all targets that need tier 1 support in ROS2.
 
 * Warnings in Helix QAC addressed
 * Code coverage according to [MC/DC](https://en.wikipedia.org/wiki/Modified_condition/decision_coverage) available
+
+## Training material recommended for contributors
+
+* Effective C++ by Scott Meyers
+* [Unit Testing and the Arrange, Act and Assert (AAA) Pattern](https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80) by Paulo Gomes
```

Comment:
@elfenpiff Feel free to add other sources here :-)

Reply:
@mossmaurice here you go:
Nicolai M. Josuttis,The C++ Standard Library
Scott Meyers,Effective Modern C++
Jeff Langr,Modern C++ Programming with Test-Driven Development: Code Better, Sleep Better
Andrei Alexandrescu,Modern C++ Design: Generic Programming and Design Patterns
Herb Sutter,Exceptional C++
Maurice Herlihy,The Art of Multiprocessor Programming
Stephan Roth, Clean C++: Sustainable Software Development Patterns and Best Practices with C++ 17
Anthony Williams,C++ Concurrency in Action

Reply:
Done